apps/rmh-web/components/SideBar.tsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+import { User } from "@/app/pt-portal/models";
+import Image from "next/image";
+import { useMemo, useState } from "react";
+
+import UsersList from "./UsersList";
+import Tab from "./Tab";
+import { useSelector } from "react-redux";
+import { selectUsers } from "@/features/usersSlice";
+
+const tabs = ["Today's", "All"];
+
+const SideBar = ({
+  isLoading,
+  users,
+}: {
+  isLoading: boolean;
+  users: User[];
+}) => {
+  const [search, setSearch] = useState("");
+  const [activeIndex, setActiveIndex] = useState(0);
+  const state = useSelector(selectUsers);
+
+  const finalUsers = useMemo(() => {
+    const userFilter = (user: User) =>
```

Comment:
this logic should probably be abstracted out as a hook and just take the dependency parameters as function